### PR TITLE
H-934, H-1025, H-1283, H-1445: Use new inference endpoint in browser plugin, add passive inference

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Update the `plugin-browser` to:
- Use the new inference endpoint added in #3630 
- Allow/track multiple inference requests
- Add passive inference option

**Removes**: the AI Python worker and the GraphQL `inferEntities` mutation from the previous implementation

## 🔍 What does this change?

As well as the above:
- extracts a shared `pluralize` function as I needed to correct a pluralization (which we might not need if we decide not to pluralize link titles such as Employed By, but it'll be useful when we need to correct others)
- updates some types to do with inference to add more useful bits of information

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] affected the execution graph, and the `turbo.json`'s have been updated to reflect this

## 🐾 Next steps

- H-1195: Have the inference feature look for existing entities to update (rather than always creating new)
- H-1284: Allow limiting passive inference to specific domains
- H-1446: Test with more complex types
- Next priorities to be determined by testing – might be better error handling (e.g. ask the model to retry validation failures), or something else

## ❓ How to test this?

1. See #3630 for instructions on running the AI TS worker locally
2. Run `yarn dev` in the plugin, then in chrome://extensions 'load unpacked' and select the `build` directory

## 📹 Demo

https://github.com/hashintel/hash/assets/37743469/0d906539-0329-46ee-a739-89cfc8af5b58

### CODE REVIEW

**Comment by nonparibus** [comment]
@TimDiekmann Do we want to persist the deprecated PY worker in the labs repo as previously discussed?

**Review by TimDiekmann — DISMISSED** [review]
I would not move the worker itself. The only interesting part in the python worker are the setup and the AI logic. The setup is heavily inspired by the official docs and it's better to look it up instead of looking at an old package and the logic was completely ported over and was improved.

The validation we used is implemented in `hash-graph-types-py`. To keep the validation I'd move that over (and the other packages).

**Review by benwerner01 — APPROVED** [review]
Thanks Ciaran, looks good to me 👍

File: libs/@local/hash-isomorphic-utils/src/pluralize.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import pluralizeLib from "pluralize";
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply libs/@local/hash-isomorphic-utils/src/pluralize.ts:1]: Agreed

File: apps/plugin-browser/src/pages/popup/popup-contents/action-center/infer-entities-action/select-types-and-infer.tsx [hunk 2]
```diff
@@ -24,36 +32,20 @@ import {
...
+    getEntityTypesQuery,
+  ).then(({ data: { queryEntityTypes } }) => {
+    return getRoots<EntityTypeRootType>(
+      /**
+       * Asserted for two reasons:
+       * 1. Inconsistencies between Graph API and hash-subgraph types
+       * 2. The function signature of getRoots asks for all fields on a subgraph, when it only needs roots and vertices
+       * @todo fix this in the Block Protocol package and then hash-subgraph
+       */
+      queryEntityTypes as Subgraph<EntityTypeRootType>,
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/plugin-browser/src/pages/popup/popup-contents/action-center/infer-entities-action/select-types-and-infer.tsx:46]: Yes – we can tidy this up soon

**Comment by TimDiekmann** [comment]
Searching in the repository for the python worker still results in very many results. I propose to remove the python worker in a dedicated PR.

**Review by nonparibus — DISMISSED** [review]

**Comment by CiaranMn** [comment]
> Turns out that my IDE cached a few results 😒 Things I found about the python worker:
> 
> * `REQUIRES_AI_WORKER_PY=true` in `.github/workflows/test.yml:88`
> * `- build-py-worker` in `.github/workflows/hash-backend-cd.yml:365`

Thank you – fixed in https://github.com/hashintel/hash/pull/3644/commits/339e057503b40dc95842547e3bb0c2734de82f87

**Review by TimDiekmann — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**